### PR TITLE
Bits needed to enforce sane disk selection in anaconda

### PR DESCRIPTION
**Pull request by dwlehman**
This goes with rhinstaller/anaconda#60

- move getRelatedDisks into blivet as suggested by both @vpodzime and @mulkieran
- tweak getDependentDevices to (optionally) work with hidden devices

### CODE REVIEW

File: blivet/devicetree.py [hunk 1]
```diff
@@ -361,28 +361,48 @@ def processActions(self, dryRun=False, callbacks=None):
...
+
+        for device in devices:
+            log.debug("checking if %s depends on %s", device.name, dep.name)
             if device.dependsOn(dep):
                 dependents.append(device)
 
         return dependents
 
+    def getRelatedDisks(self, disk):
+        """ Return a list of disks related to disk by container membership.
```
Comment (mulkieran) [inline blivet/devicetree.py]: Literally, it returns a set, not a list.

Comment (mulkieran) [inline blivet/devicetree.py]: Also the docstring talks about "container membership" but given the definition of getDependentDevices it feels like getRelatedDisks is doing something far more general than anything to do with containers, per se.

**Comment by mulkieran** [comment]
Other than the docstring, it looks fine to me.

File: blivet/devicetree.py (hunk 1 above)
Comment (dwlehman) [inline blivet/devicetree.py]: Unless I'm forgetting something, the only way for two devices to share a child is through container membership.

**Comment by vpodzime** [comment]
Looks good to me. The ``:rtype: set of :class:`~.devices.StorageDevice``` would be nice, though.

**Comment by dwlehman** [comment]
Improved docstring for getRelatedDisks

**Pull request merged**